### ISSUE TITLE
Write system test plan

### ISSUE THREAD

**Issue by mew-nsc** [debrief/pepys-import#930]
## 🐞 Overview
Produce a system test plan to verify end-to-end functionality and performance of Pepys. The system test plan should consist of a series of test steps and their expected outcomes. A completed test plan should note whether the actual result matches the expectation. 
Something like: 
|Ref | Test step | Expected Result | Pass/Fail| Comment |
|---|---|---|---|---|
| 1   | Press big red button | Nothing happens | Fail | Something happened |

Test steps could be grouped to allow this to be broken up and tested by more than one person.

Tests should verify:
- Installation instructions are sufficient to install Pepys on target system(s)
- Different workflows through the tool
  - Normal workflow for importing is clear and works
  - Invalid workflows are discouraged and don't cause invalid state
- Documentation is suitable for completing tasks in Pepys

The aim here is to catch things that wouldn't be caught by the automated unit tests. For example, we don't need to test every valid/invalid case in the import functions, but we need to make sure that we cover the typical workflows around importing and using the imported data in Pepys.

**Comment by robintw** [comment]
I agree that we probably need some manual tests, but I just wanted to mention a few of the automated tests that we do that cover some of these areas:

- We have a 'test automaton' in `test_import_cli_end_to_end.py` which runs the import CLI and randomly gives different answers to each question asked by the importer. It found quite a lot of bugs when we first implemented it - and seems to be a fairly good way of exercising the UI by giving valid and invalid answers. I've copied the docstring below, as it explains it quite well:

> These tests use an 'automaton' to do end-to-end testing of the import command-line interface.
    This automaton gives reasonable answers to each question asked by the command-line resolver:
    either choosing from the list, where a list or auto-completion list is given, or giving one of a
    predefined set of answers.

> This means, whenever the test is run, the interface is 'exercised' with a load of potential
    interactions, which can flush out various bugs. The tests use randomness in choosing which
    answers to give, so are run ten times each to get a reasonable coverage, and I have implemented
    tests for importing a REP file and a Nisida file. If the test fails then the output that will be
    shown includes the random seed used to run the test, so it can be recreated for testing
    purposes. A lot of the interactions 'go round in circles' a bit, as it gives invalid input, so
    the question is asked again, and again, until it randomly picks a valid input. Still, that is a
    good test of our invalid input handling code! Basically, if the CLI can deal with this automaton
    hammering it with crazy stuff, then it should be able to deal with a real human operating it
    sensibly!

> This is a relatively naive automaton at the moment, and doesn't do absolutely everything that is
    possible in the GUI, but it has already brought a number of bugs to my attention. These bugs are
    mostly around checking input from the user more carefully to deal with empty input, input that
    is too long, input that isn't properly parsed etc. The actual results of the import are not
    checked: if the CLI doesn't fail with an exception then the test passes.

- We have a Github Actions workflow that creates a deployable release from every commit in a PR. Once it has created the release, it then installs it and runs a simple test import. This tests the install script works, but doesn't test UI-related things like Windows shortcuts being created correctly.

**Comment by IanMayo** [comment]
**Note:** there could be some benefit in some investigation into how such a system test could be automated - to maximally exploit the value invested.

This could happen in parallel with the STP skeleton being developed.

**Comment by robintw** [comment]
As discussed in our meeting last week, here are some rough outline plans for a manual, interactive system test plan.

All of these steps use a new SQLite database specified using the `--db` option
1. Using Pepys Admin, initialise database and populate reference and metadata
2. Run an import of `uk_track.rep` in interactive mode (ie. _not_ the default importer), answering all the questions sensibly (possibly including deliberately answering a couple in an invalid way and checking it deals with it properly)
3. View the highlighted output of the import

In the maintenance GUI:
4. View the Platforms table and check that the Splendid entry exists. Use the view option to view all of the columns associated with it
5. View the Sensors table, and filter to just the sensors on the Splendid platform
5. View the Nationalities table, and delete an entry
6. View the Platform Types table and merge an entry
7. 
In the tasks GUI:
7. Create one entry for each hierarchy of the tasks: a series, a wargame and a serial. Add participants (noting that only participants for the wargame can be added to the serial etc) and dates and times, and save
8. Delete the wargame, and see that the serial is deleted

How's that for a starting point @IanMayo? Shouldn't take more than 10mins for an experienced Pepys user.

**Comment by IanMayo** [comment]
Yes, that's good @robintw .  I was also pleased with your observations on verifying the install ran correctly.  So, I guess the first step is "download release candidate".

**Comment by robintw** [comment]
Oh of course - I'll just add a few more things to the list above.

**Comment by robintw** [comment]
Do you think we can mark this as completed now @IanMayo?

**Comment by IanMayo** [comment]
I don't think the document is complete yet.  I think I (or someone) should give V1 a "formal approval".
